pyproject.toml

```diff
@@ -0,0 +1,45 @@
+
+[project]
+name = "pyimcom"
+license = {file = "LICENSE"}
+readme = "README.md"
+authors = [
+    { name = "Kaili Cao", email = "cao.1191@osu.edu" }
+]
+classifiers = [
+    "Development Status :: 4 - Beta",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Operating System :: OS Independent",
+    "Programming Language :: Python",
+]
+dynamic = ["version", "dependencies", "optional-dependencies"]
+requires-python = ">=3.11"
+
+[tool.setuptools.dynamic]
+dependencies = { file = "requirements.txt" }
```

Comment:
Dynamically adding it from `requirements.txt` is indeed better for the reasons mentioned in the PR description.